scenarios/aws/ec2/vm.go

```diff
@@ -0,0 +1,79 @@
+package ec2
+
+import (
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/components/remote"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/DataDog/test-infra-definitions/resources/aws/ec2"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+// NewVM creates an EC2 Instance and returns a Remote component.
+// Without any parameter it creates an Ubuntu VM on AMD64 architecture.
+func NewVM(e aws.Environment, name string, params ...VMOption) (*remote.Host, error) {
+	vmArgs, err := buildArgs(params...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Default missing parameters
+	if err = defaultVMArgs(e, vmArgs); err != nil {
+		return nil, err
+	}
+
+	// Resolve AMI if necessary
+	amiInfo, err := resolveOS(e, vmArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the EC2 instance
+	return components.NewComponent(*e.CommonEnvironment, e.Namer.ResourceName(name), func(c *remote.Host) error {
+		instanceArgs := ec2.InstanceArgs{
+			AMI:          amiInfo.id,
+			InstanceType: vmArgs.instanceType,
+			UserData:     vmArgs.userData,
+		}
+
+		// Create the EC2 instance
+		instance, err := ec2.NewInstance(e, name, instanceArgs, pulumi.Parent(c))
+		if err != nil {
+			return err
+		}
+
+		// Create connection
+		conn, err := remote.MakeConnection(instance.PrivateIp, amiInfo.defaultUser, e.DefaultPrivateKeyPath(), e.DefaultPrivateKeyPassword(), "")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Is `Make*` a go pattern ? I am more used to `New*` or `Create*`

Reply:
It's not a Go pattern, I do see it used in some libraries but that's it. I could be `NewConnection` but I aligned with `MakeHost`.

Reply:
The difference with `MakeHost` is that it fills an existing instance, while here you create a new one, which makes me think about `New*` for go